neuralprophet/custom_loss_metrics.py

```diff
@@ -0,0 +1,39 @@
+import torch
+from torch.nn.modules.loss import _Loss
+
+
+class PinballLoss(_Loss):
+    """Class for the PinBall loss for quantile regression"""
+
+    def __init__(self, loss_func, quantiles=None):
+        """
+        Args:
+            loss_func (torch.nn._Loss): Loss function to be used as the
+                base loss for pinball loss
+            quantiles (list): list of quantiles estimated from the model
+        """
+        super(PinballLoss, self).__init__()
+        self.loss_func = loss_func
+        self.quantiles = quantiles
+
+    def forward(self, outputs, target):
+        """
+        Computes the pinball loss from forecasts
+        Args:
+            outputs (torch.tensor): outputs from the model of dims (batch, no_quantiles, n_forecasts)
+            target (torch.tensor): actual targets of dims (batch, n_forecasts)
+
+        Returns:
+            pinball loss (float)
+        """
+        target = target.repeat(1, len(self.quantiles), 1)  # increase the quantile dimension of the targets
+        differences = target - outputs
+        base_losses = self.loss_func(outputs, target)  # dimensions - [n_batch, no. of quantiles, n_forecasts]
+        positive_losses = torch.tensor(self.quantiles).unsqueeze(dim=-1).unsqueeze(dim=0) * base_losses
+        negative_losses = (1 - torch.tensor(self.quantiles).unsqueeze(dim=-1).unsqueeze(dim=0)) * base_losses
+        pinball_losses = torch.where(differences >= 0, positive_losses, negative_losses)
+        # multiplier = torch.ones(size=(1, len(self.quantiles), 1))
```

Comment:
I can double the median weight, but the median quantile loss will be double the weight of the other quantile loss, regardless of how many quantiles there are. Alternatively, I can double the losses for all of the quantiles so that they are "scaled" to the base loss. For example, **2*(0.1 or 0.9)*bass_loss** for a 90th quantile depending on positive or negative loss. All of the quantile losses will averaged out.

Reply:
based on our conversation: let's double the median and sum over all quantiles at the end

Reply:
done